src/inmanta/env.py

```diff
@@ -69,17 +124,106 @@ def install_from_source(cls, paths: List[LocalPackagePath]) -> None:
             raise Exception("install_from_source requires at least one package to install")
         # make sure we only try to install from a local source
         explicit_paths: Iterator[LocalPackagePath] = (
-            LocalPackagePath(path=os.path.join(".", path.path), editable=path.editable) for path in paths
+            LocalPackagePath(path=os.path.join(".", path.path, ""), editable=path.editable) for path in paths
         )
         subprocess.check_call(
             [
-                cls.env_path,
+                cls.python_path,
                 "-m",
                 "pip",
                 "install",
                 *chain.from_iterable(["-e", path.path] if path.editable else [path.path] for path in explicit_paths),
             ]
         )
+        # TODO: other install methods patch pkg_resources.working_set, do we need to do so here as well?
+
+    @classmethod
+    def check(cls, in_scope: Pattern[str], constraints: Optional[List[Requirement]] = None) -> bool:
+        """
+        Check this Python environment for compatible dependencies in installed packages.
+
+        :param in_scope: A full pattern representing the package names that are considered in scope for the installed packages'
+            compatibility check. If constraints are supplied, those are always checked, regardless of this pattern.
+        :param constraints: In addition to checking for compatibility within the environment, also verify that the environment's
+            packages meet the given constraints. This is insterpreted as a list of constraints, not a list of requirements: if
+            one of the constraints' package is not installed the check succeeds.
+        """
+
+        def check_installed() -> List[str]:
+            process: subprocess.CompletedProcess = subprocess.run(
+                [
+                    cls.python_path,
+                    "-m",
+                    "pip",
+                    "check",
+                ],
+                stdout=subprocess.PIPE,
+            )
+            # returncodes 0 and 1 are expected
+            if process.returncode == 0:
+                return []
+            elif process.returncode == 1:
+                stdout: str = process.stdout.decode()
+                return [line for line in stdout.splitlines() if in_scope.fullmatch(line.split()[0])]
+            else:
+                process.check_returncode()
+            return []
+
+        def check_constraints() -> Iterator[Tuple[Requirement, version.Version]]:
+            if constraints is None:
+                return iter(())
+
+            installed_packages: Dict[str, version.Version] = {
+                name.lower(): version for name, version in get_installed_packages(cls.python_path).items()
```

Comment:
This is another fork I would very much try to avoid.

Reply:
I've updated this to use `pkg_resources.working_set` instead.

Reply:
Nice, this caused a test speedup from +- 30 min to +- 24 min.